src/main/java/uk/gov/hmcts/reform/et/syaapi/service/pdf/ET3FormService.java

```diff
@@ -0,0 +1,196 @@
+                                                               UserInfo userInfo) {
+        if (ObjectUtils.isEmpty(selectedRespondent) || ObjectUtils.isEmpty(selectedRespondent.getValue())) {
+            return StringUtils.EMPTY;
+        }
+        if (StringUtils.isNotBlank(selectedRespondent.getValue().getResponseRespondentName())) {
+            return selectedRespondent.getValue().getResponseRespondentName();
+        }
+        if (StringUtils.isNotBlank(selectedRespondent.getValue().getRespondentOrganisation())) {
+            return selectedRespondent.getValue().getRespondentOrganisation();
+        }
+        String respondentName = getRespondentNameByUserInfo(userInfo);
+
+        if (StringUtils.isBlank(respondentName)) {
+            respondentName = getRespondentNameByRespondentFirstAndLastNames(selectedRespondent);
+        }
+        return respondentName.trim();
+    }
+
+    private static String getRespondentNameByUserInfo(UserInfo userInfo) {
+        if (ObjectUtils.isEmpty(userInfo)) {
+            return StringUtils.EMPTY;
+        }
+        String respondentName = StringUtils.EMPTY;
+        if (StringUtils.isNotBlank(userInfo.getGivenName())) {
+            respondentName = userInfo.getGivenName();
+        }
+        if (ObjectUtils.isNotEmpty(userInfo) && StringUtils.isNotBlank(userInfo.getFamilyName())) {
+            respondentName = respondentName + StringUtils.SPACE + userInfo.getFamilyName();
+        }
+        return respondentName.trim();
+    }
+
+    private static String getRespondentNameByRespondentFirstAndLastNames(RespondentSumTypeItem respondent) {
+        if (ObjectUtils.isEmpty(respondent)) {
+            return StringUtils.EMPTY;
+        }
+        String respondentName = StringUtils.EMPTY;
+        if (StringUtils.isNotBlank(respondent.getValue().getRespondentFirstName())) {
+            respondentName = respondent.getValue().getRespondentFirstName();
+        }
+        if (StringUtils.isNotBlank(respondent.getValue().getRespondentLastName())) {
+            respondentName = respondentName
+                + StringUtils.SPACE
+                + respondent.getValue().getRespondentLastName();
+        }
+        return respondentName.trim();
+    }
+
+    public void generateET3WelshAndEnglishForms(String authorisation,
+                                                CaseData caseData,
+                                                RespondentSumTypeItem selectedRespondent) {
+        log.info("Generating ET3 Welsh and English Forms");
+        try {
+            DynamicFixedListType dynamicFixedListType = new DynamicFixedListType();
+            DynamicValueType dynamicValueType = new DynamicValueType();
+            dynamicValueType.setCode("SubmitRespondent");
+            dynamicValueType.setLabel(selectedRespondent.getId());
+
+            caseData.setSubmitEt3Respondent(dynamicFixedListType);
+            caseData.getSubmitEt3Respondent().setValue(dynamicValueType);
```

Comment:
Minor point - There's code in the implementation of the DynamicFixedListType class to simplify this.
```suggestion
            caseData.setSubmitEt3Respondent(DynamicFixedListType.from("SubmitRespondent", selectedRespondent.getId(), true);
```

Reply:
Updated as
caseData.setSubmitEt3Respondent(DynamicFixedListType.from("SubmitRespondent",
                                                                      selectedRespondent.getId(),
                                                                      true));